larpandoracontent/LArHelpers/LArDiscreteProbabilityHelper.h

```diff
@@ -0,0 +1,192 @@
+/**
+ *  @file   larpandoracontent/LArHelpers/LArDiscreteProbabilityHelper.h
+ *
+ *  @brief  Header file for the discrete probability helper class.
+ *
+ *  $Log: $
+ */
+#ifndef LAR_DISCRETE_PROBABILITY_HELPER_HELPER_H
+#define LAR_DISCRETE_PROBABILITY_HELPER_HELPER_H 1
+
+#include "larpandoracontent/LArObjects/LArDiscreteProbabilityVector.h"
+
+#include <algorithm>
+#include <random>
+
+namespace lar_content
+{
+
+/**
+ *  @brief  LArDiscreteProbabilityHelper class
+ */
+class LArDiscreteProbabilityHelper
+{
+public:
+    /**
+     *  @brief  Calculate P value for measured correlation coefficient between two datasets via a permutation test 
+     *
+     *  @param  t1 the first input dataset
+     *  @param  t2 the second input dataset
+     *  @param  randomNumberGenerator the random number generator to shuffle the datasets
+     *  @param  nPermutations the number of permutations to run
+     *
+     *  @return the p-value
+     */
+    template <typename T>
+    static float CalculateCorrelationCoefficientPValueFromPermutationTest(const T &t1, const T &t2, 
+        std::mt19937 &randomNumberGenerator, const size_t nPermutations);
```

Comment:
RNG presumably can't be by const reference? Here and throughout.

Reply:
It cannot sadly, the state has to be updated when numbers are thrown.